# Graded assignment 1

Mark Klik & Misja Mikkers


# Packages

We need the package `tidyverse`

In [2]:
library(tidyverse) 

-- Attaching packages --------------------------------------- tidyverse 1.2.1 --
<U+221A> ggplot2 2.2.1     <U+221A> purrr   0.2.4
<U+221A> tibble  1.4.2     <U+221A> dplyr   0.7.4
<U+221A> tidyr   0.8.0     <U+221A> stringr 1.3.0
<U+221A> readr   1.1.1     <U+221A> forcats 0.3.0
-- Conflicts ------------------------------------------ tidyverse_conflicts() --
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()


# Assignment 1

First, we generate some random data:

In [3]:
df1 <- as.data.frame(matrix(runif(20*2), ncol=20) )
head(df1)

V1        V2         V3        V4        V5        V6        V7       
1 0.4139665 0.58440043 0.5646792 0.6735486 0.1392801 0.3371608 0.5707504
2 0.5872659 0.01847601 0.7261767 0.2025068 0.7406035 0.9344503 0.1989304
  V8        V9        V10       V11       V12       V13       V14      
1 0.2736153 0.6556310 0.6081974 0.2072842 0.5243748 0.4121256 0.8939809
2 0.3271408 0.6134119 0.2691507 0.7039736 0.7872457 0.2554810 0.7515314
  V15       V16       V17       V18       V19       V20      
1 0.8725667 0.5305840 0.1609914 0.7603526 0.2643382 0.2686754
2 0.9407200 0.8349283 0.5605913 0.1577431 0.3044133 0.1641563

_df1_ is a wide dataframe. The goal of the assignment is to change _df1_ into a long dataframe _df2_. And show the first rows of the dataframe using `head(df2)`

The dataframe _df2_ should look like this:

| Key | Value      |
|-----|------------|
| V1  | 0.09413864 |
| V1  | 0.69782526 |
| V2  | 0.23460454 |
| V2  | 0.51520514 |
| V3  | 0.63181844 |
| V3  | 0.22807211 |

In [6]:
df2 <- df1 %>%
  gather(key,value)

head(df2)

key value     
1 V1  0.41396653
2 V1  0.58726586
3 V2  0.58440043
4 V2  0.01847601
5 V3  0.56467922
6 V3  0.72617674

## Assignment 2

We will generate some data:

In [7]:
df3 <- as.data.frame(read.table(header=TRUE, text='
      id        V_1       V_2                
          1  42.467573  -87.810047     
          2  42.049148  -88.273029
          3  39.110539  -90.324080
                       '))

head(df3)

id V_1      V_2      
1 1  42.46757 -87.81005
2 2  42.04915 -88.27303
3 3  39.11054 -90.32408

In [8]:
df4 <- as.data.frame(read.table(header=TRUE, text='
     V_3            V_4         id                 
      one         1h            1
      two         1h            1
      three       8h            2
      four        1h            4
'))

head(df4)

V_3   V_4 id
1 one   1h  1 
2 two   1h  1 
3 three 8h  2 
4 four  1h  4


You will have to merge these files, returning all rows and all columns from both _df3_ and _df5_. Where there are not matching values, return _NA_ for the missings. 

This should be the end result:

| id | V_1      | V_2       | V_3   | V_4 |
|----|----------|-----------|-------|-----|
| 1  | 42.46757 | -87.81005 | one   | 1h  |
| 1  | 42.46757 | -87.81005 | two   | 1h  |
| 2  | 42.04915 | -88.27303 | three | 8h  |
| 3  | 39.11054 | -90.32408 | NA    | NA  |
| 4  | NA       | NA        | four  | 1h  |



In [9]:
df5 <- full_join(df3, df4, by = "id")
df5

id V_1      V_2       V_3   V_4
1 1  42.46757 -87.81005 one   1h 
2 1  42.46757 -87.81005 two   1h 
3 2  42.04915 -88.27303 three 8h 
4 3  39.11054 -90.32408 NA    NA 
5 4        NA        NA four  1h

End of notebook